### PR TITLE
Automate build process to change version - Issue 96 

### LINKED ISSUE #96: Automate build process to...
[ ] take version number from package and store in manifest
Other?  Add here as they are identified.

### PR DESCRIPTION
**Pull request by YassineSMR**
We took the version of index.ts and sync it in other files

Fixes #96

Co-authored-by: @toto101230

### CODE REVIEW

File: autobuild.js [hunk 1]
```diff
@@ -0,0 +1,39 @@
+const fs = require('fs');
```
Comment (ear1grey) [inline autobuild.js:1]: use the ES6 import syntax - `require` will die eventually,

File: autobuild.js [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+  }
+]
+
+const readInterface = readline.createInterface({
+  input: fs.createReadStream(fromPath),
+  console: false
+});
+
+
+readInterface.on('line', function(line) {
```
Comment (ear1grey) [inline autobuild.js:26]: might it be easier / more reliable to `export` the version number and then `import` it?

**Review by ear1grey — CHANGES_REQUESTED** [review]

**Comment by toto101230** [comment]
We managed to use import but we couldn't import index because it was having problems with chrome.

**Comment by ear1grey** [comment]
Now this is finished, I think a better way to do it, so... reopened!

In TS there's a [--resolveJsonModule](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-2-9.html#new---resolvejsonmodule) compiler switch that when used allows `import` to load `.json` files (as `require`) previously did.

This means that:
 - [ ] `package.json` can be the authoritative source.
 - [ ] `version.js` can be removed
 - [ ] `index.ts` can `import` the `package.json` file for access to the version string
 - [ ] `manifest.json` can be updated by `autobuild.js` based on the version in `package.json`
 - [ ] `npm version` can become the preferred way of updating the version number (and adding tags to git).
 
 Thoughts @jacekkopecky?

**Comment by jacekkopecky** [comment]
we can also have a `version` script in package.json (not `postversion` btw) that updates `manifest.json` with the new version from `package.json` (the `version` script runs when `npm version` has updated `package.json`); then manifest.json gets committed as part of the version commit.

As an added exercise, we could also have a pre-commit git check that only allows version changes in package.json in commits with the appropriate version tag, and maybe CI could also check this.

**Pull request merged**